components/raftstore/src/store/peer_storage.rs

```diff
@@ -1404,14 +1405,11 @@ where
         }
     }
 
-    /// Evict entries from the cache.
-    pub fn evict_cache(&mut self, half: bool) {
+    /// Evict applied entries to save more memory.
+    pub fn evict_cache(&mut self) {
         if !self.cache.cache.is_empty() {
             let cache = &mut self.cache;
-            let cache_len = cache.cache.len();
-            let drain_to = if half { cache_len / 2 } else { cache_len - 1 };
-            let idx = cache.cache[drain_to].index;
-            let mem_size_change = cache.compact_to(idx + 1);
+            let mem_size_change = cache.compact_to(self.apply_state.applied_index + 1);
```

Comment:
it may cause raft to fetch entries in disk more frequently, consider using replicated_idx which you can refer to `on_raft_gc_log_tick`

Reply:
1. currently will not work. because the leader proposal flow is not controled by this. If we use replicated index, then if one peer is slow, oom will happen easily.

Reply:
Is this evict_cache running in fsm thread----Will it block fsm message processing?
If yes, then one way to reduce disk IO is that we may check if replicated_idx is far away from applied_index. If it's not far away, then compact_to replicated_idx is enough to avoid additional IO read. But if it's far away, then compact_to applied_index may be inevitable to save enough memory.

Reply:
One point, the `far away` is not a good definition. Because this is a peer level, not the whole level. Even if each peer doesn't trigger this condition, then the whole memory can still takes huge.

Reply:
Maybe, we can try a two-level like way to evict cache.
First to evict only the replicated logs. If not, then Second, evict the applied.